type-metadata/src/tests.rs

```diff
@@ -78,7 +92,7 @@ fn struct_with_generics_metadata_impl_should_work() {
 		fn type_ident() -> IdentKind {
 			IdentKind::Custom(CustomIdent {
 				name: "MyStruct",
-				namespace: Namespace(vec!["MyTestMod"]),
+				namespace: Namespace::new(vec!["MyTestMod"]),
```

Comment:
Maybe we want to check if the path segments are actually valid, e.g. `snake_case` and small letters.

Reply:
Or maybe we shouldn't because people (or macros) want to avoid that.